### PR TITLE
Remove ParameterTypeFlags::Escaping

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <https://bugs.swift.org/browse/SR-10256>, <rdar://problem/49522774>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -2424,11 +2424,26 @@ static bool isStructOrClassContext(DeclContext *dc) {
   return isa<ClassDecl>(nominal) || isa<StructDecl>(nominal);
 }
 
+static bool isEscaping(Type type) {
+  if (auto *funcType = type->getAs<AnyFunctionType>()) {
+    if (funcType->getExtInfo().getRepresentation() ==
+          FunctionTypeRepresentation::CFunctionPointer)
+      return false;
+
+    if (funcType->getExtInfo().isNoEscape())
```
Comment (xedin) [inline lib/AST/ASTPrinter.cpp]: Nit: Could be simplified down to `return !funcType->getExtInfo().isNoEscape();`

File: lib/AST/ASTPrinter.cpp [hunk 2]
```diff
@@ -3553,7 +3556,7 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
 
   void visitParenType(ParenType *T) {
     Printer << "(";
-    printParameterFlags(Printer, Options, T->getParameterFlags());
+    printParameterFlags(Printer, Options, T->getParameterFlags(), false);
```
Comment (xedin) [inline lib/AST/ASTPrinter.cpp]: Nit: Missing `/*escaping*/` before false

File: lib/AST/ASTPrinter.cpp [hunk 3]
```diff
@@ -3584,7 +3587,7 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
         visit(TD.getVarargBaseTy());
         Printer << "...";
       } else {
-        printParameterFlags(Printer, Options, TD.getParameterFlags());
+        printParameterFlags(Printer, Options, TD.getParameterFlags(), false);
```
Comment (xedin) [inline lib/AST/ASTPrinter.cpp]: Missing `/*escaping*/` before false

File: lib/AST/ASTPrinter.cpp [hunk 4]
```diff
@@ -2424,11 +2424,26 @@ static bool isStructOrClassContext(DeclContext *dc) {
...
+  return false;
+}
+
 static void printParameterFlags(ASTPrinter &printer, PrintOptions options,
-                                ParameterTypeFlags flags) {
+                                ParameterTypeFlags flags, bool escaping) {
   if (!options.excludeAttrKind(TAK_autoclosure) && flags.isAutoClosure())
     printer << "@autoclosure ";
-  if (!options.excludeAttrKind(TAK_escaping) && flags.isEscaping())
+  if (!options.excludeAttrKind(TAK_escaping) && escaping)
```
Comment (xedin) [inline lib/AST/ASTPrinter.cpp]: If the order of attributes doesn't matter here maybe `@escaping` should be printed as part of the type before printing parameters?

**Review by xedin — COMMENTED** [review]
LGTM! I have left couple of comments inline.

**Comment by slavapestov** [comment]
@xedin Thanks for the review! Follow-up commit incoming.

**Pull request merged**